### PR TITLE
Changes a few parameters in dynamic distribution

### PR DESCRIPTION
**Pull request by ShiftyRail**
## Motivation

Rounds are too gunked. Even something like 50 threat is enough for a wizard + vampires + traitors.
This aims to make rounds more peaceful.

Keys numbers :
- 90% rounds between 10 and 60 threat
- 50% of rounds between 30 and 40. Enough for a single big antag.

Another way to look at it would be to just make it have less roundstart rulesets, with higher injection delays. It has the advantage of being more readable.

## Effective changes

This changes the two parameters in the distribution. The intent is to make rounds more peaceful.
Effectively, we have : 

image

As a new distribution, compared to the old one.

image

In short : 90% of the rounds will be between 10 and 60 threat.
Can be further tweaked if needed. Of course.

### CODE REVIEW

**Comment by ShiftyRail** [comment]
If people want the details of the numbers

https://docs.google.com/spreadsheets/d/1qnQm5hDdwZoyVmBCtf6-jwwHKEaCnYa3ljmYPs7gkSE/edit#gid=0

image


## Boring shit (no need to read)

Some  maths for context. Don't need to actually read it.

Given a round of a given relative threat x, the probability to get a round with higher relative  threat (T) is given by the relation : 
``` 
p(T > x) = 0.5 - 1/pi * arcrtan((x - x0)/s))
```
With x0 and s being the parameters of the distribution curve

You can also get the probability of a round of threat between x1 and x2 by just using : 
```
P(x1 < T < x2) = P(T > x2) - P(T > x1)
```

This is what I use to get the probabilities for the next part. 

### The hell is relative threat

The relation I have gives me a number between -inf and +inf. I need to change it to a number between 0 and 100. 
The rules for this are like this.

```
x < -20 : any number between 0 and 10.
-20 < x < -10 : rule of three between 10 and 20. (-20 is 10, -10 is 20, and a linear relation in the middle)
-10 < x < -5 : rule of three between 20 and 30.
-5 < x < -2.5 : rule of three between 30 and 40.
-2.5 < x < 0 : rule of three between 40 and 50.
0 < x < 2.5 : rule of three between 50 and 60.
2.5 < x < 5 : rule of three between 60 and 70.
5 < x < 10 : rule of three between 70 and 80.
10 < x < 20 : rule of three between 80 and 90.
x > 20 : rule of three between 90 and 100.
```

**Comment by boy2mantwicethefam** [comment]
I say the current moderate approach is better because it offers an equal chance of having a peaceful round and a chaotic round, and the main problem will be that certain antags will be even more rare than they already are, and rounds would generally have one or two kinds of antags and nothing more unless either a catbeast spawns and survives long enough (and they usually don't, especially on AI rounds) or people go out of their way to destroy antag bodies, which is usually the way to deal with vampires but nobody does for other antags without intentionally trying to go for threat refunding
I will post roundstart, midround, then latejoin costs for reference

Roundstart (start of the round):
Traitor/autotraitor = 10 threat per traitor
Vampires = 25
Wizard = 30
Civil War of Casters = 45
Cult = 30
Nuke Ops = 40
Malf AI = 40
Two blob overminds = 45
Revolution = 35

Midround (middle of the round):
Autotraitor, but already mentioned before = 10 per traitor
Malf AI = 35
Ragin' Mages = 20 per mage
Nuke Ops = 35
Two blob overminds = 30
Revsquad = 45
Space Ninja = 10

Latejoin (when someone joins the round):
Traitor = 5
Ragin Mages again = 20
Ninja again = 10
Solo head revolutionary = 20

The Ssethtide has started to settle down and many of the more powerful and more costly antags aren't even allowed to roll if they do not fit the requirements based on population and threat level (lower populations require higher threat level to even have a chance to roll, otherwise they are excluded, threat level itself gravitates to around 50), as well as required enemies (mostly a non-issue, everything considers security to be a required enemy and there's almost always security)
With the upcoming "no stacking" change, round-ending antags (Cult, revolutionary, blob, nuke op, and malf) will become even rarer, and I can see Cult being directly affected negatively due to far more inexperience for a teamwork-centered gamemode
The worst that can happen will be cheap, low-threat-level-requiring, high-weight antags taking up most of the rounds, and that's **traitors** (10, perpetual 10 threat level requirement, high weight), **vampires** (cheap [25] for the general amount of vampires provided, 50 threat level for 10-14 population range, but low weight), and **wizard** (30 cost, 40 threat level cost for 15-19 range, 3 weight), and this will also mean they will suck up all the threat from the round while making midround and latejoin roles far rarer as a result due to little to no threat points left
Also better to have antags than antagstended and relying on catbeasts to naturally spawn or people destroying antag bodies without preserving their brains
Approach with caution

**Comment by ShiftyRail** [comment]
It's not really 10 per traitor. It's 10 for it to roll at all. It can spawn up to 8 traitors with 10 threat

**Comment by DeityLink** [comment]
>Even something like 50 threat is enough for a wizard + vampires + traitors.

Isn't that something we should fix by raising the requirements and costs instead of changing the formula every time?

**Comment by ShiftyRail** [comment]
I mean yes, we can change either one of those arbitrary numbers

Though maybe the best idea is to make it so spawning 8+ traitors costs more than 10 threat.
And also to have longer injection cooldowns (which is addressed in two other PRs)

**Comment by ShiftyRail** [comment]
Other one is less complicated
